src/main/java/me/youhavetrouble/purpurextras/config/PurpurConfig.java

```diff
@@ -100,6 +101,17 @@ public PurpurConfig(PurpurExtras plugin) {
             plugin.registerListener(MobNoTargetListener.class);
         }
 
+        // Generate Stonecutter Damage Filter dynamically
+        for (EntityType e: EntityType.values()) {
+            if (!e.isAlive()) { continue; }
+            //noinspection UnnecessaryToStringCall
+            String path = "settings.stonecutter-damage." + e.toString(); // Even though IDE thinks .toString() isn't necessary, it is!
```

Comment:
This will generate config line for every entity type. make it a blacklist instead. Make a hashset out of this list for fast .contains checking later

Reply:
Should I add comment(`settings.stonecutter-damage.comment`) to direct user to [this javadoc pagge](https://purpurmc.org/javadoc/org/bukkit/entity/EntityType.html)?

Advantage of making it 'generate all entity type list' is, user don't need to search internet to find out what kind of mob they want to filter damage since config option will have it already. if I change this to 'blacklist'(also making not generate full list) requires user to search javadoc to find out what kind of mob they want to 'blacklist' mobs from damaged from stonecutter.

As for hashmap, I think it would be like `HashMap<EntityType, boolean>` but I can't think of how to implement that properly without forcing user to search internet for EntityType.

Reply:
you can grab the mojang name from the namespaced name, no nned to check docs, just wiki

Reply:
> you can grab the mojang name from the namespaced name, no nned to check docs, just wiki  

Does this mean I should Use [`NamespacedKey#fromString(String)`](https://purpurmc.org/javadoc/org/bukkit/NamespacedKey.html#fromString(java.lang.String))? I've never used 'namespacedkey' for Minecraft before, so I am confused.
```java
@Nullable public static @Nullable NamespacedKey fromString(@NotNull String key)
```

Reply:
fromString() would give you Namepsacedkey, which is namespace:key, we only want the key out of namespaced key, so you'd loop over EntitType, get their key and check if [`.getKey().getKey()`  ](https://purpurmc.org/javadoc/com/destroystokyo/paper/Namespaced.html#getKey()) , which would be the key without the namespace matches

Reply:
you can see how to get the mojang name from type [here](https://github.com/YouHaveTrouble/PurpurExtras/blob/17949b0a8604c016e4aeac3e69da4ea036b51ed7/src/main/java/me/youhavetrouble/purpurextras/listeners/ForceNametaggedForRidingListener.java#L38)